### DISCUSSION TITLE
QR code scanning for iOS and Android with an embedded MAUI CameraView

### DISCUSSION THREAD

**Discussion by jp-weber** [unoplatform/uno#14826] [Show and tell]
Hello,

since I have been looking for a comfortable solution to scan the required QR codes for my app (two factor authenticator via totp), I have now opted for [MAUI embedding](https://platform.uno/docs/articles/external/uno.extensions/doc/Overview/Maui/MauiOverview.html) in order to be able to use the new project [ZXing.Net.Maui](https://github.com/Redth/ZXing.Net.Maui), since [ZXing.Net.Mobile](https://github.com/Redth/ZXing.Net.Mobile) has not been properly supported for ages. 

Test project: https://github.com/jp-weber/UnoCameraMauiTest

I am quite impressed by the functionality and embedding of the MAUI control:
![grafik](https://github.com/unoplatform/uno/assets/10946850/845a7be8-e0e0-4cd4-a843-2367d08b63ed)

Core of the test project: 
Read QR code in MAUI library and set variable in ViewModel =>
```csharp
    private void BarcodesDetected(object sender, ZXing.Net.Maui.BarcodeDetectionEventArgs e)
    {
        foreach (var barcode in e.Results)
        {
            Console.WriteLine($"Barcodes: {barcode.Format} -> {barcode.Value}");
            ViewModel.Parameter = barcode.Value;
        }
    }
```
Send the value of the variable to the recipient in the Uno project =>
```csharp
    public string Parameter
    {
        get => _parameter;
        set
        {
            if(SetProperty(ref _parameter, value) && !string.IsNullOrWhiteSpace(value))
            {
                SendParameterMessage();
            }
        }
    }

    public void SendParameterMessage()
    {
        Messenger.Send(new QRCodeScannedMessage(Parameter));
    }
```
And last the recipient in the Uno project
```csharp
        Messenger.Register<MainPageViewModel, QRCodeScannedMessage>(this, async (viewmodel, message) =>
        {
            // on Android the UI update must be set on UI Thread, else =>
            // Android.Util.AndroidRuntimeException: Only the original thread that created a view hierarchy can touch its views
            await App.MainWindow.Dispatcher.RunAsync(CoreDispatcherPriority.Normal, () =>
            {
                viewmodel.Parameter = message.Value;
            });
            
        });
```

What is your opinion about this combination to realize the required functionality?

**Comment by jeromelaban** [comment]
@jp-weber thanks for showing your solution! If MAUI embedding solves your problem, it's one way to do it for sure. Maybe there will be a different "uno native" solution in the future to do this.
